pkg/operator/ceph/controller/spec.go

```diff
@@ -84,12 +97,19 @@ fi
 # edit the logrotate file to only rotate a specific daemon log
 # otherwise we will logrotate log files without reloading certain daemons
 # this might happen when multiple daemons run on the same machine
-sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FILE"
+sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FOLDER"
 
 while true; do
-	sleep "$PERIODICITY"
-	echo "starting log rotation"
-	logrotate --verbose --force "$LOG_ROTATE_CEPH_FILE"
+	sleep "30m"
+	CURRENT_TIME_IN_SEC=$(date +%s)
+	TIME_DIFF_IN_HOUR=$(echo $((CURRENT_TIME_IN_SEC - START_TIME_IN_SEC)) | awk '{printf "%d", $1/3600'})
```

Comment:
Why don't we use bash built-in math here instead of piping to awk? It reads more cleanly.

```suggestion
	TIME_DIFF_IN_HOUR=$(( (CURRENT_TIME_IN_SEC - START_TIME_IN_SEC) / 3600 ))
```

Reply:
in built match, bash gives error when it is decimal but seems like we are good here

Reply:
You can use `3600.0` to cause bash to give a float value return.